newton/_src/sim/ik/ik_solver.py

```
@@ -0,0 +1,446 @@
+                line_search_alphas=line_search_alphas,
+                wolfe_c1=wolfe_c1,
+                wolfe_c2=wolfe_c2,
+            )
+        else:
+            raise ValueError(f"Unsupported optimizer: {optimizer}")
+
+        self.costs_expanded = self._impl.costs
+
+    def step(self, joint_q_in: wp.array2d, joint_q_out: wp.array2d, iterations=50, step_size=1.0):
+        if joint_q_in.shape != (self.n_problems, self.n_coords):
+            raise ValueError("joint_q_in has incompatible shape")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)